src/coprocessor/dag/rpn_expr/impl_op.rs

```diff
@@ -54,6 +54,21 @@ impl RpnFnLogicalOr {
     }
 }
 
+#[derive(Debug, Clone, Copy, RpnFunction)]
+#[rpn_function(args = 1)]
+pub struct RpnFnUnaryNot;
+
+impl RpnFnUnaryNot {
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        arg: &Option<i64>,
```

Comment:
TiDB currently is incompatible with MySQL 8 when decimal and floating point numbers is involved. If this function requires an `i64` input it is still easy for TiDB to fix the compatibility bug?

MySQL 8:

```
mysql> select not -0.01, not -0.0, not 0.0, not 0.01;
+-----------+----------+---------+----------+
| not -0.01 | not -0.0 | not 0.0 | not 0.01 |
+-----------+----------+---------+----------+
|         0 |        1 |       1 |        0 |
+-----------+----------+---------+----------+
1 row in set (0.01 sec)
```

On TiDB:

```
mysql> select not -0.01, not -0.0, not 0.0, not 0.01;
+-----------+----------+---------+----------+
| not -0.01 | not -0.0 | not 0.0 | not 0.01 |
+-----------+----------+---------+----------+
|         1 |        1 |       1 |        1 |
+-----------+----------+---------+----------+
1 row in set (0.00 sec)
```

Reply:
To handle this, maybe TiDB can introduce a new function called UnaryNotReal?